numerics/src/tools/NumericsMatrix.c

```diff
@@ -2891,6 +2891,200 @@ int NM_gesv_expert(NumericsMatrix* A, double *b, unsigned keep)
+      {
+        if (verbose >= 2)
+        {
+          printf("NM_posv: dense LU solve DGETRS failed. The %d-th argument has an illegal value, stopping\n", -info);
+        }
+      }
+    }
+    else
+    {
+      double* mat;
+      if (keep == NM_PRESERVE)
+      {
+        mat = NM_dWork(A, A->size0*A->size1);
+        cblas_dcopy_msan(A->size0*A->size1, A->matrix0, 1, mat, 1);
+      }
+      else
+      {
+        mat = A->matrix0;
+      }
+      DPOSV(LA_UP, A->size0, 1, mat, A->size0, b,
+            A->size0, &info);
+      if (info > 0)
+      {
+        if (verbose >= 2)
+        {
+          printf("NM_posv: Cholesky solver DPOSV failed. The %d-th diagonal element is 0\n", info);
+        }
+      }
+      else if (info < 0)
+      {
+        fprintf(stderr, "NM_posv: Cholesky solver DPOSV failed. The %d-th argument has an illegal value, stopping\n", -info);
+      }
+    }
+    break;
+  }
+
+  case NM_SPARSE_BLOCK: /* sparse block -> triplet -> csc */
+  case NM_SPARSE:
+  {
+    NSM_linear_solver_params* p = NSM_linearSolverParams(A);
+    switch (p->solver)
+    {
+    case NSM_CS_CHOLSOL:
+      numerics_printf_verbose(2,"NM_posv, using CSparse cholsol" );
+
+      if (keep == NM_KEEP_FACTORS)
+      {
+        if (!(p->dWork && p->linear_solver_data))
+        {
+          assert(!NSM_workspace(p));
+          assert(!NSM_linear_solver_data(p));
+          assert(!p->solver_free_hook);
+
+          p->solver_free_hook = &NSM_free_p;
+          p->dWork = (double*) malloc(A->size1 * sizeof(double));
+          p->dWorkSize = A->size1;
+          CSparseMatrix_factors* cs_chol_A = (CSparseMatrix_factors*) malloc(sizeof(CSparseMatrix_factors));
+          numerics_printf_verbose(2,"NM_posv_expert, we compute factors and keep it" );
+          CHECK_RETURN(CSparsematrix_chol_factorization(0, NM_csc(A),  cs_chol_A));
+          p->linear_solver_data = cs_chol_A;
+        }
+
+        numerics_printf_verbose(2,"NM_posv, we solve with given factors" );
+        info = !CSparseMatrix_chol_solve((CSparseMatrix_factors *)NSM_linear_solver_data(p), NSM_workspace(p), b);
+      }
+      else
+      {
+        numerics_printf_verbose(2,"NM_posv" );
+        info = !cs_cholsol(1, NM_csc(A), b);
+        if (info > 0)
+        {
+          printf("NM_posv: cs_cholsol failed. info = %i\n", info);
+        }
+        //DEBUG_EXPR(NV_display)
+      }
+      break;
+
+#ifdef WITH_MUMPS
+    case NSM_MUMPS:
+    {
+      if (verbose >= 2)
+      {
+        printf("NM_gesv: using MUMPS\n" );
+      }
+      if (!NM_MUMPS_id(A)->job || (NM_MUMPS_id(A)->job == -2))
+      {
+        /* the mumps instance is initialized (call with job=-1) */
+        NM_MUMPS_set_control_params(A);
+        NM_MUMPS(A, -1);
+        NM_MUMPS_set_verbosity(A, verbose);
+        NM_MUMPS_set_icntl(A, 24, 1); // Null pivot row detection
+        NM_MUMPS_set_cntl(A, 5, 1.e20); // Fixation, recommended value
+      }
+      NM_MUMPS_set_problem(A, b);
+
```

Comment:
I think we should call `NM_MUMPS_set_sym(A, 2)` here to set the proper symmetry value? In `NM_MUMPS_set_control_params` the `sym` value is set to `0`. In the MUMPS manual, `sym` can have 3 values: `0` for (unsymmetric) general matrices, `1` for symmetric positive definite, `2` for general symmetric.

Reply:
Yes I left the same code than for NM_gesv. This has to be done. 
I was waiting to discuss with @bremond to know it we need another interface for LDLT or if we can put in NM_posv. 
For interior point method we need is explicitly  and not a simple Cholesky 
Thanks for the comment